src/writer/proposal.ts

```diff
@@ -217,6 +208,16 @@ export async function verify(body): Promise<any> {
     capture(e);
     return Promise.reject('failed to check proposals limit');
   }
+
+  const bodyLengthLimit = await getLimit(`space.${await getSpaceType(space)}.body_limit`);
+  if (msg.payload.body.length > bodyLengthLimit) {
+    return Promise.reject(`proposal body length can not exceed ${bodyLengthLimit} characters`);
+  }
+
+  const choicesLimit = await getLimit(`space.${await getSpaceType(space)}.choices_limit`);
+  if (msg.payload.choices.length > choicesLimit) {
+    return Promise.reject(`number of choices can not exceed ${choicesLimit}`);
```

Comment:
I think we should call `getSpaceType` on top level instead of four times, we can then pass `spaceType` to `getSpaceProposalsLimits` instead of `space`

Reply:
maybe we should also have `isEcosystemSpace` on top and pass type to `getSpaceProposalsLimits`